portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,19 +14,35 @@
 
 package com.google.sps.servlets;
 
+import java.util.ArrayList;
+import java.util.List;
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
-@WebServlet("/data")
+@WebServlet("/random-welcome")
 public class DataServlet extends HttpServlet {
 
+  private List<String> welcomes;
+
+  @Override
+  public void init() {
+    welcomes = new ArrayList<>();
```

Comment:
I recommend inlining the definition and making the member `final`.

`private List<String> welcomes = new ArrayList<>();
`
This is a simple mechanism of guaranteeing that this member will never be `null`.